test/API/H5_api_link_test.c

```diff
@@ -897,6 +897,12 @@ test_create_hard_link_invalid_params(void H5_ATTR_UNUSED *params)
                 PART_ERROR(H5Lcreate_hard_across_files);
             }
 
+            if (H5Fclose(ext_file_id) < 0) {
```

Comment:
This file handle needed to be closed before attempting to remove the file on Windows